cumulusci/core/config/project_config.py

```diff
@@ -485,6 +491,42 @@ def _check_keychain(self):
                 + "config.set_keychain(keychain) before accessing orgs"
             )
 
+    def get_repo_from_url(self, url):
+        repo_owner, repo_name = url.split("/")[3:5]
+        if repo_name.endswith(".git"):
+            repo_name = repo_name[:-4]
+        gh = self.get_github_api(repo_owner, repo_name)
+        repo = gh.repository(repo_owner, repo_name)
+
+        return (repo, repo_owner, repo_name)
+
+    def get_ref_for_dependency(self, repo, dependency, include_beta=None):
+        release = None
+        if "ref" in dependency:
+            ref = dependency["ref"]
+        else:
+            if "tag" in dependency:
+                try:
+                    # Find the github release corresponding to this tag.
+                    release = repo.release_from_tag(dependency["tag"])
+                except NotFoundError:
+                    raise DependencyResolutionError(
+                        f"No release found for tag {dependency['tag']}"
+                    )
+            else:
+                release = find_latest_release(repo, include_beta)
+            if release:
+                ref = repo.tag(
+                    repo.ref("tags/" + release.tag_name).object.sha
+                ).object.sha
+            else:
+                self.logger.info(
+                    f"No release found; using the latest commit from the {repo.default_branch} branch."
+                )
+                ref = repo.branch(repo.default_branch).commit.sha
+
+        return (release, ref)
```

Comment:
This doesn't have to happen right now, but I think we should be moving in the direction of making BaseProjectConfig resolve dependencies by delegating to Source objects (like we already do for resolving cross-project flows) so that we can have different Source implementations for different version control systems.